spec/policies/order/delivery_status_policy_spec.rb

```diff
@@ -0,0 +1,122 @@
+require 'spec_helper'
+
+describe Order::DeliveryStatusPolicy do
+
+  let(:market_org)     { create(:organization, :market) }
+  let!(:market)        { create(:market, organization: market_org) }
+  let(:market_manager) { create(:user, :market_manager, managed_markets: [market], organizations: [market_org]) }
+
+  let(:supplier)       { create(:user, :supplier) }
+  let(:buyer)          { create(:user, :buyer) }
+
+  let(:order)          { create(:order, :with_items, market: market) }
+
+  let(:delivery_status_policy) { Order::DeliveryStatusPolicy.new(user, order) }
+
+  before do
+    allow(order).to receive(:undelivered_for_user?).with(user) { undelivered }
+  end
+
+  permissions :mark_delivered? do
+    context 'as a market manager' do
+      let(:user) { market_manager }
+
+      context 'when order is not yet delivered' do
+        let(:undelivered) { true }
+
+        it 'returns true if order not direct delivery' do
+          allow(order).to receive_message_chain('delivery.delivery_schedule.direct_to_customer?') { false }
+          expect(delivery_status_policy.mark_delivered?).to be true
+        end
+
+        it 'returns true if order is direct delivery' do
+          allow(order).to receive_message_chain('delivery.delivery_schedule.direct_to_customer?') { true }
+          expect(delivery_status_policy.mark_delivered?).to be true
+        end
```

Comment:
The [pundit rspec style](https://github.com/varvet/pundit#policy-specs) is to use their `permit` helper:
````
expect(described_class).to permit(user, order)
````
It might read better that the policy 'allows access if order is direct delivery' instead of 'returns true if order is direct delivery'. I says these comments just to get a consistent style for pundit tests started, you're not doing anything wrong.

Reply:
Gotcha. Don't stop with the input, I love it!